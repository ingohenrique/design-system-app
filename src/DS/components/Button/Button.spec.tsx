import {
  describe,
  expect,
  it,
  vi
} from "vitest"

import {fireEvent, render, screen} from "@testing-library/react"
import { DS_Button } from "."
import { FaSignInAlt } from "react-icons/fa"

describe("Components/Button", () => {
  it("should render with children text", () => {
    render(<DS_Button state="default">Entrar</DS_Button>)

    expect(screen.getByText("Entrar")).toBeTruthy()

  })

  it("should render with right icon", () => {
    const handleClick = vi.fn();

    render(
      <DS_Button
        state="outlined"
        rightIcon={<FaSignInAlt />}
        onClick={handleClick}
      >
        Entrar
      </DS_Button>
    )

    fireEvent.click(screen.getByText("Entrar"));

    expect(handleClick).toHaveBeenCalledTimes(1);
    expect(screen.getByTestId("right-icon")).toBeTruthy();
    expect(screen.getByText("Entrar")).toBeTruthy();
    })

  it("should render with left icon", () => {
    const handleClick = vi.fn();

    render(
      <DS_Button
        state="outlined"
        leftIcon={<FaSignInAlt />}
        onClick={handleClick}
      >
        Entrar
      </DS_Button>
    )

    fireEvent.click(screen.getByText("Entrar"));

    expect(handleClick).toHaveBeenCalledTimes(1);
    expect(screen.getByTestId("left-icon")).toBeTruthy();
    expect(screen.getByText("Entrar")).toBeTruthy();
    })

  it("should take a snapshot", () => {
    const { asFragment } = render(
      <DS_Button state="default">Entrar</DS_Button>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  })