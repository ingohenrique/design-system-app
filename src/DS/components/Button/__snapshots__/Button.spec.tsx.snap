// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components/Button > should take a snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-egkSDF TUmSN"
    state="default"
  >
    <span
      class="sc-blHHSb hhkiqo"
    >
      Entrar
    </span>
  </button>
</DocumentFragment>
`;
